src/engine/permission-handler.ts

```diff
@@ -0,0 +1,86 @@
+export interface Permissions {
+  [to: string]: {
+    [from: string]: {
+      allow: boolean,
+      hash: string,
+    }
+  }
+}
+
+export interface IPermissionProvider {
+  confirm(message: string): Promise<{ allow: boolean, remember: boolean }>
+  /** Clear the permission object */
+  clear(): void
```

Comment:
isn't this part should be in the `permissionHandler` interface? (cause this is directly modifying the permissions list

Reply:
Yes, I think `clear` should be in the Handler. I'll change that.